cores/arduino/WInterrupts.c

```diff
@@ -250,7 +250,7 @@ void detachInterrupt(uint32_t pin)
  * External Interrupt Controller NVIC Interrupt Handler
  */
 #if defined(__SAMD51__)
-void InterruptHandler(uint32_t i)
+void InterruptHandler(__attribute__((unused)) uint32_t i)
```

Comment:
hmm, I am not entirely sure you want to declare the function as unused or the parameter as unused.

Reply:
While I don't recall, I'm pretty sure the warning was about the **_parameter_** being unused.
From the following, you can see that the parameter is hidden by the loop variable:

https://github.com/adafruit/ArduinoCore-samd/blob/12c45064377642fd49a7c1c8de6d16d99947e3d0/cores/arduino/WInterrupts.c#L253-L259